src/main/java/cn/nukkit/player/handler/LoginPacketHandler.java

```diff
@@ -17,14 +17,18 @@
 import com.nukkitx.protocol.bedrock.packet.PlayStatusPacket;
 import lombok.extern.log4j.Log4j2;
 
-import java.util.Objects;
 import java.util.function.Consumer;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
  * @author Extollite
  */
 @Log4j2
 public class LoginPacketHandler implements BedrockPacketHandler {
+
+    private final Pattern pattern = Pattern.compile("^[aA-zZ\\s\\d_]{3,16}+$");
```

Comment:
still isn't static...

Reply:
> still isn't static...

Excuse me, I turned out to be inattentive